doc/reference/trd104-syscalls.md

```diff
@@ -571,6 +560,28 @@ userspace needs to read or write to a buffer held by the kernel, it
 MUST first regain access to it by calling the corresponding Read-Write
 Allow.
 
+The return variants for Read-Write Allow system calls are `Failure
+with 2 u32` and `Success with 2 u32`.  In both cases, `Argument 0`
+contains an address and `Argument 1` contains a length. When a driver
+implementing the Read-Write Allow system call returns a failure
+result, it MUST return the same address and length as those that were passed 
+in the call. When a driver implementing the Read-Write Allow system call
+returns a success result, the returned address and length MUST be those
+that were passed in the previous call, unless this is the first call.
+On the first successful invocation of a particular Read-Write Allow system 
+call, an driver implementation MUST return address 0 and size 0.
+
+Note that these requirements are typically on capsule code, and are not
+enforced by the core kernel. Because capsules may have bugs or not be fully
+trust, userspace SHOULD NOT rely on the fact that drivers follow these rules as
+described, especially if the capsule code is not independently checked and tested.
+Instead, userspace SHOULD employ additional checks to ensure that
+drivers behave as specified, and implement proper error handling as needed.
+The core kernel MAY enforce a subset of these
+specified rules in the future.
+
+The buffer number specifies which buffer this is. A driver may
+support multiple allowed buffers.
```

Comment:
I believe these lines should still be further up, closer to where the arguments to subscribe are listed.

Reply:
Yeah, this confused me.

Reply:
Yeah I think this was inadvertently cut off. I think the whole section needs a re-org; the points are made in strange orders.